mangopy/mosaic.py

```diff
@@ -0,0 +1,304 @@
+        dlon = lon - lon0 
+        dlat = lat - lat0 
+        a = np.sin(dlat/2)**2 + np.cos(lat0) * np.cos(lat) * np.sin(dlon/2)**2
+        c = 2 * np.arcsin(np.sqrt(a)) 
+        # Radius of earth in kilometers is 6371
+        km = 6371* c
+        return km
+
+
+    def get_nearest_index(self,site,background_grid,time):
+
+        rewrite_file = False
+        regrid_file = os.path.join(self.mangopy_path,'regrid_image_index.h5')
+        if rewrite_file:
+            os.remove(regrid_file)
+
+        try:
+            with h5py.File(regrid_file,'r') as f:
+                nearest_idx = f[site['name']][:]
+        except:
+            
+            flat_grid = np.array([background_grid[0].ravel(),background_grid[1].ravel()]).T
+            lon_arr = background_grid[0,0,:]
+            lat_arr = background_grid[1,:,0]
+
+            # get site lat/lon arrays
+            __, lat, lon, __ = self.get_data(site,time)
+
+            flat_lat = lat.ravel()
+            flat_lon = lon.ravel()
+            flat_idx = np.arange(len(flat_lat))
+
+            flat_idx = flat_idx[np.isfinite(flat_lat)]
+            flat_lon = flat_lon[np.isfinite(flat_lat)]
+            flat_lat = flat_lat[np.isfinite(flat_lat)]
+            flat_points = np.array([flat_lon,flat_lat]).T
+
+            fov = flat_points[ConvexHull(flat_points).vertices].T
+            # print(fov)
+            center_lon = site['lon']
+            if center_lon<0:
+                center_lon += 360.
+            # find points west of site
+            fovW = fov[:,fov[0,:]<center_lon]
+            # find points east of site
+            fovE = fov[:,fov[0,:]>center_lon]
+            # find longitude limits for each latitude row in the grid
+            limits = []
+            for f in [fovW,fovE]:
+                f = f[:,np.argsort(f[1,:])]
+                limits.append(np.interp(lat_arr,f[1,:],f[0,:],left=np.nan,right=np.nan))
+            # create flag array identifying points in grid within fov
+            flags = np.all(np.array([lon_arr>=limits[0][:,None],lon_arr<=limits[1][:,None]]),axis=0)
+
+            # find index of image cell that is closest to each grid cell in the fov
+            nearest_idx = np.full(background_grid[0].shape,np.nan)
+            nearest_idx[flags] = interpolate.griddata(flat_points,flat_idx,flat_grid[flags.ravel()],method='nearest')
+
+            with h5py.File(regrid_file, 'a') as f:
+                f.create_dataset(site['name'], data=nearest_idx, compression='gzip', compression_opts=1)
+
+        return nearest_idx
+
+
+    def grid_mosaic(self,time,grid,hiarchy):
+        grid_img = []
+        truetime = []
+        for site in self.site_list:
+
+            print(site['name'])
+            # get data
+            try:
+                img, __, __, tt = self.get_data(site,time)
+                truetime.append(tt)
+            except (OSError, IOError, ValueError) as e:
+                print(e)
```

Comment:
Suggestion: 
print("Exception: %s' % str(e))

Reply:
Looks good, that's more informative!